rclcpp/include/rclcpp/experimental/buffers/ring_buffer_implementation.hpp

```diff
@@ -51,6 +53,7 @@ class RingBufferImplementation : public BufferImplementationBase<BufferT>
     if (capacity == 0) {
       throw std::invalid_argument("capacity must be a positive, non-zero value");
     }
+    TRACEPOINT(construct_ring_buffer, static_cast<const void *>(this), capacity_);
```

Comment:
Just curious and for the knowledge, why are we proving buffer construction and clear? that would only happens initialization and finalization, so not really performance critical prove point? these are needed to tracing information?

Reply:
Thanks for your review. As you noted, `construct_ring_buffer` and `ring_buffer_clear` are not currently used for measurements. On the other hand, these tracepoints have the following use case.

For example, suppose that for some reason there is a high latency from publish to callback. This could be due to a large queue capacity. In this case, we might want to know how many messages are stored in the queue. The queue capacity is now the only information available in the construct, and it is sufficient to record this information once during the construct.

Also, `ring_buffer_clear` is useful for calculating the number of messages stuck in the queue. If there is no tracepoint, it is difficult to accurately calculate the number of messages in the queue immediately after clear is called.
Of course, clear is not called now, but we think it should be added in anticipation of future developments.

In summary, these two tracepoints provide the information needed to track the state of the queue.

Reply:
Thanks for sharing the insight, that was really helpful. lgtm!